src/data/finsupp/basic.lean

```diff
@@ -1483,6 +1483,50 @@ end
 
 end comap_domain
 
+section option
+
+/-- Restrict a finitely supported function on `option α` to a finitely support function on `α`. -/
+def some [has_zero M] (f : option α →₀ M) : α →₀ M :=
+f.comap_domain option.some (by tidy)
+
+@[simp] lemma some_apply [has_zero M] (f : option α →₀ M) (a : α) :
+  f.some a = f (option.some a) := rfl
+
+@[simp] lemma zero_sum [has_zero M] : (0 : option α →₀ M).some = 0 :=
+by { ext, simp, }
+
+@[simp] lemma add_sum [add_comm_monoid M] (f g : option α →₀ M) : (f + g).some = f.some + g.some :=
+by { ext, simp, }
+
+@[simp] lemma single_none_some [has_zero M] (m : M) : (single none m : option α →₀ M).some = 0 :=
+by { ext, simp, }
+
+@[simp] lemma single_some_some [has_zero M] (a : α) (m : M) :
+  (single (option.some a) m : option α →₀ M).some = single a m :=
+by { ext b, simp [single_apply], }
+
+lemma sum_option [add_comm_monoid M] [add_comm_monoid N]
+  (f : option α →₀ M) (b : option α → M →+ N) :
+  f.sum (λ a, b a) = b none (f none) + f.some.sum (λ a, b (option.some a)) :=
+begin
+  apply induction_linear f,
+  { simp, },
+  { intros f₁ f₂ h₁ h₂,
+    rw [finsupp.sum_add_index, h₁, h₂, add_sum, finsupp.sum_add_index],
+    simp only [add_monoid_hom.map_add, pi.add_apply, finsupp.coe_add],
+    abel,
+    all_goals { simp, }, },
+  { rintros (_|a) m; simp, }
+end
+
+lemma sum_option' [semiring R] [add_comm_monoid M] [module R M]
```

Comment:
This only needs `distrib_mul_action`, doesn't it?

Reply:
`smul_add_hom` wants the full `module` argument.